core/modules/cart/store/getters.ts

```diff
@@ -58,6 +58,9 @@ const getters: GetterTree<CartState, RootState> = {
       code: state.platformTotals.coupon_code,
       discount: state.platformTotals.discount_amount
     }
+  },
+  isVirtualCart (state) {
```

Comment:
hey @ananth-iyer 
we don't need this flas in state, please just refactor this getter to something like this
```
isVirtualCart (state) {
  return state.cartItems.every((itm) => {
      return itm.type_id === 'downloadable' || itm.type_id === 'virtual' // check for downloadable & virtual products
    })
}
```

and then we don't need code in `action.ts`, `index.ts`, `CartState.ts` 

also we don't have to worry if any other action would change cartItems and doesn't recalculate flag, because this getter will always tell the truth.

Reply:
I have refractor the code in latest commit, review it